### PR TITLE
Make canadarm's xacro file configurable

### LINKED ISSUE #14: Update canadarm demo to control the arm
Enhancements
- Add a configurable model
- Fix meshes rendering
- Make it work with Rviz and Gazebo simulateneously

### PR DESCRIPTION
**Pull request by ana-GT**
Hi :) . I had some issues/errors while trying to use the Canadarm demo. I fixed said issues with this PR (+ [one other PR in the demos repo](https://github.com/space-ros/demos/pull/21)). The changes are the following:

1. Missing collision meshes: Commented them out and use the .stl versions of the visual meshes instead.
2. Create a xacro file for the canadarm to replace the current urdf. The xacro file has as an argument the xyz/rpy location for the arm base, making it easier to configure the pose.
3. Create a macro for the gazebo's ros2 control interface. The macro has an argument for the type of hardware interface. The original urdf had an **effort** interface, which didn't work. Changed it to be **position** , which works fine.  

Tested with these changes (and the aforementioned demos's PR) and it works as expected.  Tested in a ROS Humble environment.

https://github.com/space-ros/simulation/assets/555379/37d77884-ada8-42cc-b4c2-7934d7b2bc05

### CODE REVIEW

**Comment by Bckempa** [comment]
@quarkytale I tagged you for review since you're our current demo expert but if you don't have room on your plate for this we can find someone else, thanks!

**Comment by quarkytale** [comment]
Hi @ana-GT, thanks for this contribution, really helps streamlining the demos. Question: how are you testing this, using `ros2 run rqt_joint_trajectory_controller rqt_joint_trajectory_controller`? I'm not able to control the joints, `rqt_joint_trajectory_controller` throws `rqt_qui` module not found. If that's the correct way to test, I'll recheck my installation, otherwise what am I missing?

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]:
```suggestion
```

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (quarkytale) [inline models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro:1]: Fine with this being added but the `models/canadarm/urdf/SSRMS_Canadarm2.urdf` should not be removed as canadarm_moveit_config depends on that. Note moveit doesn't recognize `package://` paths hence they were hardcoded.

**Comment by ana-GT** [comment]
Hi, apologies for the late reply, I was dead busy with a project. Will address the suggested changes and will reply soon on regards to the testing procedure I used (I think I created a rospy node to send motion commands, but not quite sure, I'll get back to you on that shortly)

**Comment by quarkytale** [comment]
No worries thanks for working on this! Ideally it would be better to use the `gz_ros2_control` interface itself since we're aiming to update that in this PR.

**Comment by quarkytale** [comment]
Update: I could control the arm using `ros2 action` command (thanks to Ezra), planning to approve this after changes are addressed.

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (ana-GT) [reply models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]: Deleted the commented out line, per reviewer's suggestion

File: models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro
Comment (ana-GT) [reply models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro]: Deleted the commented out line, per reviewer's suggestion

**Comment by ana-GT** [comment]
> Update: I could control the arm using `ros2 action` command (thanks to Ezra), planning to approve this after changes are addressed.

Happy to hear that! I pushed a commit that addressed the comments above (delete the commented out lines).

I only have one more change I'd like to make, but first wanted to check: So, the robot loads fine in Gazebo, but if you try loading it in **Rviz, the user gets an error that the meshes are not being found.** From what I can tell, the reason is that we are using:
```
<mesh filename="$(find simulation)/models...
```

If we replace it with:

```
<mesh filename="package://simulation/models...
```
**then the arm shows up fine both in Gazebo and Rviz2**. Is there any reason why not use **package** in the urdf.xacro? I think I read in the comments above that MoveIt might have trouble with it. If that is not longer the case, I would be happy to add that last change to this PR. Let me know! 
gz_rviz

PS.- Tested this time in the **demos_ws** docker.

**Comment by quarkytale** [comment]
Apologies for the late reply, I was travelling. I think the moveit_config is using `models/canadarm/urdf/SSRMS_Canadarm2.urdf` directly and I'm assuming you're launching RViz with the new `models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro` file you added right? If so, feel free to make the package path edit. We're reworking the moveit demo anyway.

**Comment by ana-GT** [comment]
> Apologies for the late reply, I was travelling. I think the moveit_config is using `models/canadarm/urdf/SSRMS_Canadarm2.urdf` directly and I'm assuming you're launching RViz with the new `models/canadarm/urdf/SSRMS_Canadarm2.urdf.xacro` file you added right? If so, feel free to make the package path edit. We're reworking the moveit demo anyway.

I am just launching rviz and then using the **/robot_description** topic that is being published from robot_state_publisher.


In any case, I updated the xacro file with **"package://xxx"** instead of **"$(find xxx)"** and now you can also see the robot in Rviz: 
test_canadarm


To test, as usual:

```ros2 launch canadarm canadarm.launch.py```

and then I just run any of the 3 services to move the arm:

```
ros2 service call /open_arm std_srvs/srv/Empty {}
ros2 service call /close_arm std_srvs/srv/Empty {}
ros2 service call /random_arm std_srvs/srv/Empty {}
```

**Comment by quarkytale** [comment]
Thanks for iterating. This is great I could capture the same movement in RViz as in Gazebo!
image

**Review by quarkytale — APPROVED** [review]

**Comment by quarkytale** [comment]
@Bckempa this is ready to be merged!

**Review by ivanperez-keera — APPROVED** [review]

**Comment by ana-GT** [comment]
If there is anything else needed for this PR to be merged, kindly let me know :) . Just checking as I just built a space-ros docker image and the canadarm simulation is failing because this PR is not yet merged. Thanks!

**Comment by quarkytale** [comment]
Hey @ana-GT I think merging is blocked because commits need to be signed.

**Comment by ana-GT** [comment]
@quarkytale  I see. I amended my last commit to be signed (didn't have my GPG keys set up yet :disappointed:  , but have it now :smiley:  ).  Would this PR require to create a new verified commit with all the info from the previous commits?

**Comment by Bckempa** [comment]
Sorry @ana-GT, we tried to get this handled at the last tech meeting but we hit a few snags with some people out and others on work computers without their keys.

Two things:

1) To follow the project contribution procedure this all needs to be squashed into a commit ending in the issue number before being merged, see @ivanperez-keera's [contributing rules](https://github.com/space-ros/space-ros/blob/main/CONTRIBUTING.md) for details. The web UI forces a fast-forward rather than a merge after a squash so that has to be done on a dev machine. Feel free to do that yourself, otherwise someone else will when we get the permissions fixed.

2) To my knowledge there has never been discussion of requiring GPG signing on commits. @EzraBrooks is this new or an error in the repo's permissions configuration?

**Comment by EzraBrooks** [comment]
I've disabled that protection. No idea how it got enabled.

**Pull request merged**
